### ISSUE TITLE
SA1/SADX character animation issues

### ISSUE THREAD

**Issue by PiKeyAr** [ItsEasyActually/SA1_Chars#11]
This isn't an issue with this mod, but since it concerns character models I thought you might be interested. There are several character animation problems in the game, here are some of them:

1. Knuckles' pushing animation is too fast.
2. Something's wrong with Tails' "jumping while holding something" animations, either one of them isn't playing at all, or they are swapped.
3. When Tails gets a water bubble he plays the drowning animation.
4. Knuckles' running animation appears choppy, I heard it could be related to the rudimentary "motion blur" effect with disabled transparency, but it looks weird in its original form.
5. Big climbs ladders really fast, but this could've been intentional.
6. Gamma's pushing animation speed is also a bit weird when he starts pushing, although this may also have been intentional.

**Comment by ItsEasyActually** [comment]
Easiest solution here will probably be to port over their original AnimData table values and double check some of their actions as well. I'll take a look at them sometime soon. This will be good for another side project I have planned.

**Comment by KayaneRF** [comment]
7. [𝘒𝘯𝘶𝘤𝘬𝘭𝘦𝘴’ 𝘷𝘪𝘤𝘵𝘰𝘳𝘺 𝘢𝘯𝘪𝘮𝘢𝘵𝘪𝘰𝘯 𝘭𝘢𝘤𝘬𝘪𝘯𝘨 𝘮𝘢𝘬𝘦 𝘢 𝘧𝘪𝘴𝘵.](https://www.dropbox.com/s/2zgkmeue6cigicr/victory_anim.mp4?dl=0)

–𝘒𝘢𝘺𝘢𝘯𝘦

**Comment by DisapprovingOwl** [comment]
8. Characters' "walking while holding something" animations always play at a single speed, rather than playing at a variable speed depending on the analog stick input like the regular walking animations.
9. Very occasionally, the characters' slowest walking animation freaks out and plays at a super-fast speed regardless of the analog stick input. Not sure exaclty what causes this - not been able to reliably reproduce it. Have only noticed it in Chao Gardens so far, though that may be because those are the places where I'm most likely to be walking at slowest speed.

**Comment by ItsEasyActually** [comment]
They use a table of values per each animation along with a play speed. I have no idea why they would be playing at higher speeds, but I've seen similar issues with the top speed run playing very slowly. There might be a way to fix it, but I don't know it off the top of my head.

**Comment by DisapprovingOwl** [comment]
Fair play - thanks for clarifying. I think it's an issue in the original game rather than a DX-specific issue, anyway (though can't double-check due to my Dreamcast's disc motor being busted).

In a similar vein:

10. In situations where the character freezes during a transition, Tails' tails don't freeze along with him. This is easiest to see when running through the Chao Race door in the Station Square Chao Garden.

**Comment by ItsEasyActually** [comment]
That's because his tails are animated independently. I'm sure there's some way to make them pause, but I'm not sure if it's something I'd look into directly fixing. Mainly due to it being part of the morph handling which I'm still not 100% sure how it works.

**Comment by SPEEPSHighway** [comment]
I'm not sure how many characters have a "throw while holding something in midair" animation, it's hard to tell. Either way, here's some more points about animations.

I'm not sure if this still happens in the PC version but I've noticed that when certain characters get hurt (Knuckles/Amy) they loop the hurt animation before landing, whereas Sonic stays on the last frame and flows better into the animation he does when he hits the ground.

Amy's dress has a habit of flying up in certain animations (Shaking a tree, walking etc.). It doesn't happen in the AutoDemo and looks pretty weird, the only reason I can think of it happening for other than being a bug is it sort of helps against clipping after she lands from a jump, but that still happens when she starts running again anyway.

In the original Japanese version the characters are animated while holding objects. It looks pretty stupid, but it could always be an optional feature or something. Maybe there's room for those (mostly, the trampolines use a variant) unused landing animations as well, as long as they don't override the walking animation like they do in the AutoDemo. I mean, Tails, Knuckles and Amy cycle through the same animation IDs as Sonic after landing but it doesn't look much different because they're just faster clones of the standing animation.

**Comment by DisapprovingOwl** [comment]
Couple of animation issues in cutscenes rather than gameplay, both from Sonic's story. I know the cutscenes are a little janky in general, but these two examples really stood out to me as errors rather than just side effects of the slightly crude nature of how the cutscenes are handled:

- In the cutscene after defeating the Egg Hornet, just after Sonic says his first line of dialogue he makes a waving motion with his right hand. It looks like he's supposed to be doing his iconic "finger-wagging" pose from the classic games - however, his hand stays open rather than switching to the "finger-wagging" version (used in some other cutscenes, including his final cutscene and the one where he finds Knuckles/Eggman in Super Sonic's story), meaning at present he just flails his palm weirdly towards Eggman instead.
- In the cutscene after Sky Chase Act 1, when Sonic is falling through the air he's stuck in his running animation rather than any kind of tumbling/falling animation. It's possible they just left it like this as it was too small to see clearly in its original resolution, but it sticks out like a sore thumb playing the game in 1080p! The animation he uses when caught in a wind tunnel (such as the entrance to Windy Valley from the Mystic Ruins) would probably be a much better fit here.

**Comment by PiKeyAr** [comment]
I don't know about the Egg Hornet cutscene, but it seems pretty obvious to me that the running animation after Sky Chase 1, along with the high-pitched noise accompanying it is a throwback to classic Sonic games. I think even in Generations Classic Sonic often uses his walking animation when falling. That particular cutscene never seemed odd to me, I think it's just intentionally comical.

When I created this issue what I had in mind was problems coming from purely technical errors, like wrong animation speed or animations that are obviously incorrect (Tails' bubble grab). The changes you're suggesting have little to do with those things, they are closer to design changes that would require modifying the programming of the cutscenes. Up to ItsEasyActually of course, but I'd say that's quite out of scope for this mod.

**Comment by DisapprovingOwl** [comment]
Fair point - can appreciate that those aren't technical errors as such, so no worries if that sort of change isn't considered in-scope for this mod. Will keep my eyes peeled for any further in-game animation errors.

**Comment by DisapprovingOwl** [comment]
OK, think I've found a legitimate technical issue here: 

11. During the animation where Big holds a fish above his head after successfully catching it, if you have the green Life Belt upgrade it clips behind his white chest fur. 

My Dreamcast's busted so can't double-check, but looking at videos this bug may have also occurred in the original game.
